pkg/utils/defaulter/clusterresourceplacement_test.go

```diff
@@ -0,0 +1,315 @@
+						PlacementType: fleetv1beta1.PickNPlacementType,
+						TopologySpreadConstraints: []fleetv1beta1.TopologySpreadConstraint{
+							{
+								TopologyKey:       "kubernetes.io/hostname",
+								MaxSkew:           ptr.To(int32(fleetv1beta1.DefaultMaxSkewValue)),
+								WhenUnsatisfiable: fleetv1beta1.DoNotSchedule,
+							},
+						},
+					},
+					Strategy: fleetv1beta1.RolloutStrategy{
+						Type: fleetv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &fleetv1beta1.RollingUpdateConfig{
+							MaxUnavailable:           ptr.To(intstr.FromString("%15")),
+							MaxSurge:                 ptr.To(intstr.FromString("%15")),
+							UnavailablePeriodSeconds: ptr.To(15),
+						},
+						ApplyStrategy: &fleetv1beta1.ApplyStrategy{
+							Type: fleetv1beta1.ApplyStrategyTypeClientSideApply,
+						},
+					},
+					RevisionHistoryLimit: ptr.To(int32(10)),
+				},
+			},
+		},
+		"ClusterResourcePlacement with nil Tolerations fields": {
+			obj: &fleetv1beta1.ClusterResourcePlacement{
+				Spec: fleetv1beta1.ClusterResourcePlacementSpec{
+					Policy: &fleetv1beta1.PlacementPolicy{
+						PlacementType: fleetv1beta1.PickNPlacementType,
+						Tolerations: []fleetv1beta1.Toleration{
+							{
+								Key:   "key",
+								Value: "value",
+							},
+						},
+					},
+					Strategy: fleetv1beta1.RolloutStrategy{
+						Type: fleetv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &fleetv1beta1.RollingUpdateConfig{
+							MaxUnavailable:           ptr.To(intstr.FromString("%15")),
+							MaxSurge:                 ptr.To(intstr.FromString("%15")),
+							UnavailablePeriodSeconds: ptr.To(15),
+						},
+						ApplyStrategy: &fleetv1beta1.ApplyStrategy{
+							Type: fleetv1beta1.ApplyStrategyTypeClientSideApply,
+						},
+					},
+					RevisionHistoryLimit: ptr.To(int32(10)),
+				},
+			},
+			wantObj: &fleetv1beta1.ClusterResourcePlacement{
+				Spec: fleetv1beta1.ClusterResourcePlacementSpec{
+					Policy: &fleetv1beta1.PlacementPolicy{
+						PlacementType: fleetv1beta1.PickNPlacementType,
+						Tolerations: []fleetv1beta1.Toleration{
+							{
+								Key:      "key",
+								Value:    "value",
+								Operator: corev1.TolerationOpEqual,
+							},
+						},
+					},
+					Strategy: fleetv1beta1.RolloutStrategy{
+						Type: fleetv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &fleetv1beta1.RollingUpdateConfig{
+							MaxUnavailable:           ptr.To(intstr.FromString("%15")),
+							MaxSurge:                 ptr.To(intstr.FromString("%15")),
+							UnavailablePeriodSeconds: ptr.To(15),
+						},
+						ApplyStrategy: &fleetv1beta1.ApplyStrategy{
+							Type: fleetv1beta1.ApplyStrategyTypeClientSideApply,
+						},
+					},
+					RevisionHistoryLimit: ptr.To(int32(10)),
+				},
+			},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			SetDefaultsClusterResourcePlacement(tt.obj)
+			if diff := cmp.Diff(tt.obj, tt.wantObj); diff != "" {
```

Comment:
```suggestion
			if diff := cmp.Diff(tt.wantObj, tt.obj); diff != "" {
```
should match with the order in the error msg,

**mismatch (-want +got)**